lib/chutney/TorNet.py

```diff
@@ -1323,11 +1358,12 @@ def getNodeDirInfoStatusPattern(self, dir_format):
         elif desc:
             return r'^router ' + nickname + " "
         elif md:
+            if ed25519_key:
             # Not yet implemented, see #33428
-            # r'^id ed25519 " + ed25519_identity (end of line)
-            # needs ed25519-identity from #30642
-            # or the existing keys/ed25519_master_id_public_key
-            return None
+                return r'^id ed25519 ' + re.escape(ed25519_key)
+            else:
+                return None
+            # Return None if ed25519_id not found
```

Comment:
Cleanup: we usually put the comments above the code.

We use comments to explain what the code does when the program runs. Or why we wrote the code.
We don't repeat what is written in the code already.

```suggestion
                # If there is no ed25519_id, then we can't search for it
                return None
```